#22586 Add Romanian language

v613 opened the pull request:
I'm excited to announce the completion of the issue #22586, where I have added Romanian translations for several key modules.

With this update, the following modules now have Romanian translations:
- [x] cmp
- [x] core
- [x] data
- [x] debug
- [x] erd
- [x] ext
- [x] model
- [x] net
- [x] registry
- [x] tasks
- [x] team
- [x] ui

By expanding our language support, we make DBeaver more user-friendly and convenient for Romanian-speaking individuals.

v613 committed:
[l18n][ro] translate core module

References:
https://github.com/dbeaver/dbeaver/issues/22586

Commit:
[l18n][ro] translate cmp module

Commit:
[l18n][ro] translate team module

Commit:
fixup! [l18n][ro] translate team module

I forgot to add the link to the issue

Reference:
https://github.com/dbeaver/dbeaver/issues/22586

Commit:
[l18n][ro] translate registry module

References:
https://github.com/dbeaver/dbeaver/issues/22586

Commit:
fixup! [l18n][ro] translate core module

Commit:
[l18n][ro] translate core module part2

In previous commit I forgot to include the link of issue
Refer to commit: 4380b034247cbe5eaa5716f661f0e281d1553feb

References:
https://github.com/dbeaver/dbeaver/issues/22586

Commit:
[l18n][ro] translate erd module

Refernces:
https://github.com/dbeaver/dbeaver/issues/22586

Commit:
[l18n][ro] translate debug module

References:
https://github.com/dbeaver/dbeaver/issues/22586

Commit:
[l18n][ro] translate net module

References:
https://github.com/dbeaver/dbeaver/issues/22586

Commit:
[l18n][ro] translate model module

References:
https://github.com/dbeaver/dbeaver/issues/22586

Commit:
[l18n][ro] translate data module

References:
https://github.com/dbeaver/dbeaver/issues/22586

Commit:
[l18n][ro] translate tasks module

References:
https://github.com/dbeaver/dbeaver/issues/22586

Commit:
[l18n][ro] translate ui module

References:
https://github.com/dbeaver/dbeaver/issues/22586

Commit:
[l18n][ro] translate ext module

References:
https://github.com/dbeaver/dbeaver/issues/22586

LonwoLonwo commented:
Hello @v613 

Thanks for this great work!
We will look at it.

LonwoLonwo commented on plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/core/CoreResources_ro_RO.properties:
```diff
@@ -0,0 +1,514 @@
+dialog_version_update_downloader_title = Actualizator de versiuni
+dialog_version_update_error_cannot_check_version = Nu se pot ob\u0163ine informa\u0163ii despre versiune. Poate c\u0103 conexiunea la re\u0163ea este \u00EEntrerupt\u0103?
+dialog_version_update_ignore_version = Nu afi\u015Fa\u0163i pentru versiunea {0}
+dialog_version_update_n_a = N / A
```
```suggestion
dialog_version_update_n_a = N/A
```

LonwoLonwo replied:
No need for extra spaces here.

LonwoLonwo commented on plugins/org.jkiss.dbeaver.data.transfer.ui/OSGI-INF/l10n/bundle_ro_RO.properties:
```diff
@@ -0,0 +1,11 @@
+command.org.jkiss.dbeaver.core.import.data.description = Importa\u0163i date din tabelele bazei de date
+
+pref.page.name.datatransfer = Transfer de date
+keyword.org.jkiss.dbeaver.preferences.datatransfer = transfer de date transfer de rezerv\u0103 tip de mapare lungime \u00EEnlocui\u0163i numele coloanelor
```
```suggestion
keyword.org.jkiss.dbeaver.preferences.datatransfer = datatransfer transfer fallback mapping type length replace columns names de date de rezerv\u0103 tip de mapare lungime \u00EEnlocui\u0163i numele coloanelor
```

LonwoLonwo replied:
It is better to leave English keywords and add localization to them

v613 replied:
Solved in [85e8673](https://github.com/dbeaver/dbeaver/pull/22696/commits/85e8673e1afaa96c2617b4865112e8f262df311f)

v613 committed:
Update plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/core/CoreResources_ro_RO.properties

Co-authored-by: Anastasiya <45152336+LonwoLonwo@users.noreply.github.com>

Commit:
[l18n][ro] restore original keyword's value

Co-authored-by: Anastasiya
<45152336+LonwoLonwo@users.noreply.github.com>

References:
https://github.com/dbeaver/dbeaver/pull/22696#discussion_r1478599968

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.mysql/OSGI-INF/l10n/bundle_ro_RO.properties:
```diff
@@ -0,0 +1,363 @@
+meta.org.jkiss.dbeaver.ext.mysql.model.MySQLUser.host.name=Masc\u0103 gazd\u0103
+meta.org.jkiss.dbeaver.ext.mysql.model.MySQLUser.maxConnections.name=Max conexiuni
+meta.org.jkiss.dbeaver.ext.mysql.model.MySQLUser.maxUserConnections.name=Max conexiuni utilizator
+meta.org.jkiss.dbeaver.ext.mysql.model.MySQLUser.sslType.name=tip SSL
```
```suggestion
meta.org.jkiss.dbeaver.ext.mysql.model.MySQLUser.sslType.name=Tip SSL
```

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.mysql/OSGI-INF/l10n/bundle_ro_RO.properties:
```diff
@@ -0,0 +1,363 @@
+meta.org.jkiss.dbeaver.ext.mysql.model.MySQLEvent.sqlMode.name=Modul SQL
+meta.org.jkiss.dbeaver.ext.mysql.model.MySQLEvent.starts.name=\u00CEncepe
+meta.org.jkiss.dbeaver.ext.mysql.model.MySQLEvent.ends.name=Se termin\u0103
+meta.org.jkiss.dbeaver.ext.mysql.model.MySQLEvent.status.name=stare
```
```suggestion
meta.org.jkiss.dbeaver.ext.mysql.model.MySQLEvent.status.name=Stare
```

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.mysql/OSGI-INF/l10n/bundle_ro_RO.properties:
```diff
@@ -0,0 +1,363 @@
+meta.org.jkiss.dbeaver.ext.mysql.tasks.MySQLToolTableRepairSettings.useFRM.name = Folosi\u0163i FRM
+
+meta.org.jkiss.dbeaver.ext.mysql.tasks.MySQLToolWithStatus$ToolStatus.object.name = Numele tabelului
+meta.org.jkiss.dbeaver.ext.mysql.tasks.MySQLToolWithStatus$ToolStatus.messageType.name = stare
```
```suggestion
meta.org.jkiss.dbeaver.ext.mysql.tasks.MySQLToolWithStatus$ToolStatus.messageType.name = Stare
```

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.mysql/OSGI-INF/l10n/bundle_ro_RO.properties:
```diff
@@ -0,0 +1,363 @@
+org.jkiss.dbeaver.task.category.mysqlTool.description = Instrumente de baz\u0103 de date MySQL
+task.mysqlDatabaseBackup.name = Dump MySQL
+task.mysqlDatabaseBackup.description = Sarcina de export a bazei de date MySQL
+task.mysqlDatabaseRestore.name = restaurare MySQL
```
```suggestion
task.mysqlDatabaseRestore.name = Restaurare MySQL
```

v613 committed:
Update plugins/org.jkiss.dbeaver.ext.mysql/OSGI-INF/l10n/bundle_ro_RO.properties

Co-authored-by: Anastasiya <45152336+LonwoLonwo@users.noreply.github.com>

v613 committed:
Update plugins/org.jkiss.dbeaver.ext.mysql/OSGI-INF/l10n/bundle_ro_RO.properties

Co-authored-by: Anastasiya <45152336+LonwoLonwo@users.noreply.github.com>

v613 committed:
Update plugins/org.jkiss.dbeaver.ext.mysql/OSGI-INF/l10n/bundle_ro_RO.properties

Co-authored-by: Anastasiya <45152336+LonwoLonwo@users.noreply.github.com>

v613 committed:
Update plugins/org.jkiss.dbeaver.ext.mysql/OSGI-INF/l10n/bundle_ro_RO.properties

Co-authored-by: Anastasiya <45152336+LonwoLonwo@users.noreply.github.com>

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.postgresql/OSGI-INF/l10n/bundle_ro_RO.properties:
```diff
@@ -0,0 +1,839 @@
+meta.org.jkiss.dbeaver.ext.postgresql.model.PostgreJobSchedule.name.name=Nume
+meta.org.jkiss.dbeaver.ext.postgresql.model.PostgreJobSchedule.description.name=Descriere
+meta.org.jkiss.dbeaver.ext.postgresql.model.PostgreJobSchedule.enabled.name=Activat
+meta.org.jkiss.dbeaver.ext.postgresql.model.PostgreJobSchedule.start.name=start
```
```suggestion
meta.org.jkiss.dbeaver.ext.postgresql.model.PostgreJobSchedule.start.name=Start
```

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.postgresql/OSGI-INF/l10n/bundle_ro_RO.properties:
```diff
@@ -0,0 +1,839 @@
+
+driver.postgresql.description=Driver standard PostgreSQL
+driver.edb.description=Driver standard EDB
+driver.timescale.description=Driver JDBC de timp
```
```suggestion
driver.timescale.description=Driver JDBC Timescale
```

LonwoLonwo replied:
This is the database name. No need to translate it.

v613 replied:
My bad

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ext.postgresql/OSGI-INF/l10n/bundle_ro_RO.properties:
```diff
@@ -0,0 +1,839 @@
+driver.cockroach.description=Driver JDBC CockroachDB
+driver.gcloudpg.description=Driver Google Cloud SQL PostgreSQL JDBC
+driver.cratedb.description=Driver CrateDB JDBC
+driver.materialize.description=Materializa\u0163i driverul JDBC
```
So, I'm confused. Is the word "driver" translatable to the Romanian language or not? You translated it here, but not for other drivers' names.

v613 replied:
No. In context of DB the word **driver** cannot be translated. I will correct it shortly.

LonwoLonwo replied:
Okay

v613 replied:
Solved in [10f1ea4](https://github.com/dbeaver/dbeaver/pull/22696/commits/10f1ea478192d4bd7ebd5d3c9229a3c08be55b85)

LonwoLonwo commented on plugins/org.jkiss.dbeaver.model/OSGI-INF/l10n/bundle_ro_RO.properties:
```diff
@@ -0,0 +1,170 @@
+meta.org.jkiss.dbeaver.model.navigator.DBNProjectDatabases.name.name = Nume
+meta.org.jkiss.dbeaver.model.navigator.DBNLocalFolder.name.name = Nume
+meta.org.jkiss.dbeaver.model.navigator.DBNResource.nodeName.name = Nume
+meta.org.jkiss.dbeaver.model.navigator.DBNResource.resourcePath.name = cale
```
```suggestion
meta.org.jkiss.dbeaver.model.navigator.DBNResource.resourcePath.name = Cale
```

LonwoLonwo commented on plugins/org.jkiss.dbeaver.model/OSGI-INF/l10n/bundle_ro_RO.properties:
```diff
@@ -0,0 +1,170 @@
+meta.org.jkiss.dbeaver.model.navigator.DBNResource.resourcePath.name = cale
+meta.org.jkiss.dbeaver.model.navigator.DBNResource.resourceLocation.name = Loca\u0163ie
+meta.org.jkiss.dbeaver.model.navigator.DBNResource.resourceLastModified.name = Modificat ultima dat\u0103
+meta.org.jkiss.dbeaver.model.navigator.DBNResource.resourceSize.name = m\u0103rimea
```
```suggestion
meta.org.jkiss.dbeaver.model.navigator.DBNResource.resourceSize.name = M\u0103rimea
```

LonwoLonwo commented on plugins/org.jkiss.dbeaver.tasks.ui.view/src/org/jkiss/dbeaver/tasks/ui/internal/TaskUIViewMessages_ro_RO.properties:
```diff
@@ -0,0 +1,69 @@
+db_tasks_tree_column_controller_add_descr_project = Proiect de sarcin\u0103
+
+db_tasks_view_filtered_tree_text_error_message = Execu\u0163ii sarcini: introduce\u0163i o parte a mesajului de eroare
+db_tasks_view_column_controller_add_name_time = timpul sistemului
```
```suggestion
db_tasks_view_column_controller_add_name_time = Timpul sistemului
```

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/internal/UIConnectionMessages_ro_RO.properties:
```diff
@@ -0,0 +1,309 @@
+controls_client_homes_panel_label_id = ID
+controls_client_homes_panel_label_name = Nume
+controls_client_homes_panel_label_path = Cale
+controls_client_homes_panel_label_product_name = numele produsului
```
```suggestion
controls_client_homes_panel_label_product_name = Numele produsului
```

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ui.editors.connection/src/org/jkiss/dbeaver/ui/internal/UIConnectionMessages_ro_RO.properties:
```diff
@@ -0,0 +1,309 @@
+dialog_edit_driver_info = Informa\u0163ie
+dialog_edit_driver_driver = Driver
+dialog_edit_driver_library = Bibliotec\u0103
+dialog_edit_driver_path = cale
```
```suggestion
dialog_edit_driver_path = Cale
```

LonwoLonwo commented on plugins/org.jkiss.dbeaver.tasks.native/OSGI-INF/l10n/bundle_ro_RO.properties:
```diff
@@ -0,0 +1 @@
+#Generated by ResourceBundle Editor (http://essiembre.github.io/eclipse-rbe/)
```
No need in this empty file. Please, remove it.

v613 replied:
OK, but all the others are present here, so I thought it was necessary to create this empty file.

https://github.com/dbeaver/dbeaver/tree/devel/plugins/org.jkiss.dbeaver.tasks.native/OSGI-INF/l10n

LonwoLonwo replied:
No. It was incorrect before. And we need to stop repeating this wrong behavior now.

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/internal/ResultSetMessages_ro_RO.properties:
```diff
@@ -0,0 +1,463 @@
+controls_resultset_ref_menu_no_references=<No References>
+controls_resultset_ref_menu_references=<Table References>
+controls_resultset_ref_menu_no_associations=<No Foreign Keys>
+controls_resultset_ref_menu_associations=<Table Foreign Keys>
```
You can localize it too, if you want.

v613 replied:
At the moment I would prefer to keep them

v613 committed:
Update plugins/org.jkiss.dbeaver.ext.postgresql/OSGI-INF/l10n/bundle_ro_RO.properties

Co-authored-by: Anastasiya <45152336+LonwoLonwo@users.noreply.github.com>

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/internal/SQLEditorMessages_ro_RO.properties:
```diff
@@ -0,0 +1,389 @@
+source_viewer_show_partitions_ddl_text = Afi\u015Fa\u0163i parti\u0163iile
+source_viewer_show_partitions_ddl_tip = Afi\u015Fa\u0163i tabelele parti\u0163ionate DDL
+
+sql_editor_action_clear = clar
```
```suggestion
sql_editor_action_clear = Сlar
```

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/internal/UINavigatorMessages_ro_RO.properties:
```diff
@@ -0,0 +1,273 @@
+pref_page_projects_settings_description = Set\u0103rile resurselor/directoarelor proiectului DBeaver
+
+ui_navigator_loading_text_loading = Se \u00EEncarc\u0103
+ui_properties_category_information = informa\u0163ie
```
```suggestion
ui_properties_category_information = Informa\u0163ie
```

LonwoLonwo commented on plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/internal/UIMessages_ro_RO.properties:
```diff
@@ -0,0 +1,94 @@
+
+ui_actions_context_search_name = C\u0103utare contextual\u0103
+
+ui_properties_tree_viewer__to_default =  la implicit
```
```suggestion
ui_properties_tree_viewer__to_default = \ la implicit
```

LonwoLonwo replied:
Because original is:
`ui_properties_tree_viewer__to_default = \ to default`

LonwoLonwo commented on test/org.jkiss.dbeaver.ext.generic.test/OSGI-INF/l10n/bundle_ro_RO.properties:
```diff
@@ -0,0 +1 @@
+#Generated by ResourceBundle Editor (http://essiembre.github.io/eclipse-rbe/)
```
Please remove empty file.

v613 replied:
Solved in [dc41b1b](https://github.com/v613/dbeaver/commit/dc41b1b8db11db37af504a75957881ef7145b9a8)

LonwoLonwo commented on test/org.jkiss.dbeaver.ext.greenplum.test/OSGI-INF/l10n/bundle_ro_RO.properties:
```diff
@@ -0,0 +1,21 @@
+#Generated by ResourceBundle Editor (http://essiembre.github.io/eclipse-rbe/)
+
+datasource.greenplum.description = Surs\u0103 de date Greenplum
```
I think the file pass is incorrect here. We do not have localizations in our tests folder.

This is correct one: plugins/org.jkiss.dbeaver.ext.greenplum/OSGI-INF/l10n/bundle.properties

v613 replied:
Solved in [dc41b1b](https://github.com/v613/dbeaver/commit/dc41b1b8db11db37af504a75957881ef7145b9a8)

LonwoLonwo commented:
And the last one. 
It is probably better to name all files shortly, like bundle_ro.properties. Without country code. Or do you have any particular reasons for it?

v613 commented:
> And the last one. It is probably better to name all files shortly, like bundle_ro.properties. Without country code. Or do you have any particular reasons for it?

@LonwoLonwo the only reason is to have the country's flag, as in the case of Portuguese (Brazil).
![image](https://github.com/dbeaver/dbeaver/assets/15879258/536aede1-686e-4a64-8178-51624a3d8af7)

LonwoLonwo commented:
Yes, but Brazilian Portuguese is different from just Portuguese. 
But it looks like it is not your case.

Commit:
Correct driver's name

Co-authored-by: Anastasiya <45152336+LonwoLonwo@users.noreply.github.com>

References:
https://github.com/dbeaver/dbeaver/pull/22696#discussion_r1484086955

Commit:
Uppercase first word

Co-authored-by: Anastasiya <45152336+LonwoLonwo@users.noreply.github.com>

References:
  https://github.com/dbeaver/dbeaver/pull/22696#discussion_r1484082458
  https://github.com/dbeaver/dbeaver/pull/22696#discussion_r1484086955
  https://github.com/dbeaver/dbeaver/pull/22696#discussion_r1484108277
  https://github.com/dbeaver/dbeaver/pull/22696#discussion_r1484108665
  https://github.com/dbeaver/dbeaver/pull/22696#discussion_r1484133769
  https://github.com/dbeaver/dbeaver/pull/22696#discussion_r1484139118
  https://github.com/dbeaver/dbeaver/pull/22696#discussion_r1484140605
  https://github.com/dbeaver/dbeaver/pull/22696#discussion_r1484156172
  https://github.com/dbeaver/dbeaver/pull/22696#discussion_r1484158187
  https://github.com/dbeaver/dbeaver/pull/22696#discussion_r1484160065

Commit:
Remove wrong generated files

Co-authored-by: Anastasiya <45152336+LonwoLonwo@users.noreply.github.com>

References:
  https://github.com/dbeaver/dbeaver/pull/22696#discussion_r1484163125

Commit:
fixup! Remove wrong generated files

v613 commented:
> Yes, but Brazilian Portuguese is different from just Portuguese. But it looks like it is not your case.

@LonwoLonwo I apologize, I did not understand, is it necessary or preferable to rename. I'd rather not rename so we have a clean git history (I mean git's possible problems with rename detection)

LonwoLonwo commented:
Please rename the files. 
I didn't see any Git problems here with the same issue, so you also can do it. https://github.com/dbeaver/dbeaver/pull/21768

Commit:
Rename `bundle_ro_RO.properties` to `bundle_ro.properties`

Co-authored-by: Anastasiya <45152336+LonwoLonwo@users.noreply.github.com>

References:
  https://github.com/dbeaver/dbeaver/pull/22696#issuecomment-1936101706
  https://github.com/dbeaver/dbeaver/issues/22586

LonwoLonwo commented:
Sorry, I wasn't specific enough. You need to rename all files, not just bundles, and all others like somethingMessages_ro_RO.properties too.

Commit:
Remove country code "RO" from filenames

References:
  https://github.com/dbeaver/dbeaver/pull/22696#issuecomment-1937078685

LonwoLonwo commented on plugins/org.jkiss.dbeaver.registry/OSGI-INF/l10n/bundle_ro.properties:
```diff
@@ -0,0 +1,71 @@
+language.ja.label    = Japonez\u0103
+language.ko.label    = Coreean\u0103
+language.pt_BR.label = Portughez\u0103 (BR)
+language.ro_RO.label = Rom\u00E2n\u0103
```
```suggestion
language.ro.label = Rom\u00E2n\u0103
```

LonwoLonwo committed:
Merge branch 'devel' into l18n/ro_RO-localization

v613 committed:
Update plugins/org.jkiss.dbeaver.registry/OSGI-INF/l10n/bundle_ro.properties

Co-authored-by: Anastasiya <45152336+LonwoLonwo@users.noreply.github.com>

ShadelessFox commented on plugins/org.jkiss.dbeaver.registry/OSGI-INF/l10n/bundle.properties:
```diff
@@ -28,6 +28,7 @@ language.ja.label=Japanese
 language.uk.label=Ukrainian
 language.ko.label=Korean
 language.uk.label=Ukrainian
+language.ro_RO.label=Romanian
```
This should be
```suggestion
language.ro.label=Romanian
```

Additionally, this won't do anything. In order to be able to change the language from DBeaver, you need to register Romanian in a list of supported languages here:
https://github.com/dbeaver/dbeaver/blob/93b491375bee5996407e5fa412b04562730f365b/plugins/org.jkiss.dbeaver.registry/plugin.xml#L25-L38

Add the following line at the end:
```diff
         <language code="es" label="%language.es.label"/>
         <language code="tw" label="%language.tw.label"/>
         <language code="uk" label="%language.uk.label"/>
+        <language code="ro" label="%language.ro.label"/>
     </extension>
```

v613 commented:
Excuse me, could you please explain the meaning of **wait for response** label ?
Did I miss something to do ?

LonwoLonwo commented:
Hello @v613 

Thanks for noticing. I removed it.
We are waiting now for the second reviewer.

ShadelessFox commented:
My review is marked as resolved but there are no recent commits. Please see https://github.com/dbeaver/dbeaver/pull/22696#pullrequestreview-1875020412

Commit:
fix ro_RO bundle

References:
- https://github.com/dbeaver/dbeaver/pull/22696#discussion_r1486045057

v613 commented:
Hello @ShadelessFox
I apologize for my previous oversight. The changes has been pushed now.
Thank you for your understanding.

MashaKorax committed:
Merge branch 'devel' into l18n/ro_RO-localization

MashaKorax commented:
verified
